### PR TITLE
Zel2flux 171

### PR DESCRIPTION
**Pull request by w2vy**
This branch has a large number of zel -> flux changes.
The RPCs maintain backwards compatibility
I did not change any file names, so the target builds still create zelcashd etc
I figured that would better be a second step, since that will impact things outside code.

Please look these changes over closely, there may be places I should not have changed, possibly for historical reasons 

closes 171 closes 175

### CODE REVIEW

**Review by blondfrogs — CHANGES_REQUESTED** [review]
Lets make sure to update those names that i commented on. I didn't comment on all of them but please change them all to the new naming scheme.

**Review by blondfrogs — CHANGES_REQUESTED** [review]
@w2vy Can we try to use __func__ instead of hardcoding the function name 

`return rebuildfluxnodedb(params, fHelp, "rebuildfluxnodedb");` could be 

`return rebuildfluxnodedb(params, fHelp, __func__);`

**Comment by w2vy** [comment]
JA>
Looks like we are changing 
/home/main/.flux/determ_zelnodes
to 
/home/main/.flux/determ_fluxnodes
which we can't do if the files already exist, we have to perform a rename if it already exists. 
lets make sure to not rename it, and then we can do the renaming of this folder in a different PR.

TM> It looks like I only did that in one place, fixed

**Review by blondfrogs — APPROVED** [review]

File: src/init.cpp [hunk 1]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             MilliSleep(10);
     }
 
-    fZelnode = GetBoolArg("-zelnode", false);
+    fFluxnode = GetBoolArg("-fluxnode", false);
```
Comment (TheTrunk) [inline src/init.cpp:1878]: should be -zelnode

File: src/init.cpp [hunk 2]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
+        if (fFluxnode) {
+            return InitError("Fluxnode can be run only on Linux");
         }
     #endif
 
-    if (fZelnode) {
+    if (fFluxnode) {
         LogPrintf("IS FLUXNODE\n");
-        strZelnodeAddr = GetArg("-zelnodeaddr", "");
+        strFluxnodeAddr = GetArg("-fluxnodeaddr", "");
```
Comment (TheTrunk) [inline src/init.cpp:1893]: backwards compatilibty for -zelnode

File: src/init.cpp [hunk 3]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
-        LogPrintf(" addr %s\n", strZelnodeAddr.c_str());
+        LogPrintf(" addr %s\n", strFluxnodeAddr.c_str());
 
-        if (!strZelnodeAddr.empty()) {
-            CService addrTest = CService(strZelnodeAddr);
+        if (!strFluxnodeAddr.empty()) {
+            CService addrTest = CService(strFluxnodeAddr);
             if (!addrTest.IsValid()) {
-                return InitError("Invalid -zelnodeaddr address: " + strZelnodeAddr);
+                return InitError("Invalid -fluxnodeaddr address: " + strFluxnodeAddr);
```
Comment (TheTrunk) [inline src/init.cpp:1900]: backwards compatilibty for -zelnode

File: src/init.cpp [hunk 4]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
             }
         }
 
-        std::string strHash = GetArg("-zelnodeoutpoint", "");
+        std::string strHash = GetArg("-fluxnodeoutpoint", "");
```
Comment (TheTrunk) [inline src/init.cpp:1904]: backwards compatilibty for -zelnode

File: src/init.cpp [hunk 5]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
         uint256 hash = uint256S(strHash);
-        int index = GetArg("-zelnodeindex", -1);
+        int index = GetArg("-fluxnodeindex", -1);
```
Comment (TheTrunk) [inline src/init.cpp:1906]: backwards compatilibty for -zelnode

File: src/init.cpp [hunk 6]
```diff
@@ -1875,75 +1875,75 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
+        if (fluxnodeOutPoint.IsNull()) {
+            return InitError(_("Invalid fluxnode outpoint data. assign fluxnodeoutpoint and fluxnodeindex."));
         }
 
-        activeZelnode.deterministicOutPoint = zelnodeOutPoint;
+        activeFluxnode.deterministicOutPoint = fluxnodeOutPoint;
 
-        strZelnodePrivKey = GetArg("-zelnodeprivkey", "");
-        if (!strZelnodePrivKey.empty()) {
+        strFluxnodePrivKey = GetArg("-fluxnodeprivkey", "");
```
Comment (TheTrunk) [inline src/init.cpp:1916]: backwards compatilibty for -zelnode

File: src/init.cpp [hunk 7]
```diff
@@ -1960,29 +1960,29 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
         } else {
 
-             if (FindBenchmarkPath("zelbenchd", strPath)) {
-                  LogPrintf("Found zelbenchd in %s\n",strPath);
+             if (FindBenchmarkPath("fluxbenchd", strPath)) {
```
Comment (TheTrunk) [inline src/init.cpp:1963]: should be zelbenchd, above we are starting fluxbenchd

File: src/init.cpp [hunk 8]
```diff
@@ -1960,29 +1960,29 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
...
+                  LogPrintf("Found fluxbenchd in %s\n",strPath);
              } else {
                  return InitError("Failed to find benchmark application");
              }
 
-             if (FindBenchmarkPath("zelbench-cli", strPath)) {
-                  LogPrintf("Found zelbench-cli in %s\n",strPath);
+             if (FindBenchmarkPath("fluxbench-cli", strPath)) {
```
Comment (TheTrunk) [inline src/init.cpp:1969]: should be zelbench-cli

File: src/util.cpp [hunk 9]
```diff
@@ -691,9 +691,9 @@ boost::filesystem::path GetConfigFile()
     return pathZelcashConfigFile;
 }
 
-boost::filesystem::path GetZelnodeConfigFile()
+boost::filesystem::path GetFluxnodeConfigFile()
 {
-    boost::filesystem::path pathConfigFile(GetArg("-znconf", "zelnode.conf"));
+    boost::filesystem::path pathConfigFile(GetArg("-znconf", "fluxnode.conf"));
```
Comment (TheTrunk) [inline src/util.cpp:696]: compatilibty for zelnode.conf

**Review by TheTrunk — COMMENTED** [review]
@w2vy @blondfrogs please see comments for a quick bug fix

File: src/init.cpp (hunk 6 above)
Comment (w2vy) [reply src/init.cpp:1916]: Should this code look for both zelnodeprivkey and fluxnodeprivkey?
Maybe only using zelnode if fluxnode isn't found?

Comment (w2vy) [reply src/init.cpp:1916]: I just reverted the change for now

**Comment by w2vy** [comment]
I made these changes in zel2flux-171a and will issue a Pull Request

**Pull request merged**